actionmailbox/lib/action_mailbox/errors.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require "active_support/actionable_error"
+require "rails/command"
+
+module ActionMailbox
+  # Generic base class for all Action Mailbox exceptions.
+  class Error < StandardError; end
+
+  # Raised when we detect that Action Mailbox has not been initialized.
+  class SetupError < Error
```

Comment:
Something seems odd to me about the abstraction here. The fact that we need to wrap this in a new exception and then call the message on that, seems needlessly roundabout. Don't really have a suggestion for now though.

Reply:
Maybe turn the API around "treat as actionable"? We can define the actions dynamically like:

```ruby
ActionDispatch::ActionableExceptions.on ActiveRecord::StatementInvalid, /#{InboundEmail.table_name}/ do |error| # Or anything responding to #=== for the check...
  # Treat this body as the class of an "actionable error". Implementation-wise,
  # we can monkey patch the current error object.
  action "Setup Action Mailbox" do
    Rails::Command.invoke("action_mailbox:install")
  end
end
```

Reply:
Can you check the current interface? I turned it around after the RailsConf feedback I received.